internal/bminventory/inventory.go

```diff
@@ -1679,7 +1681,12 @@ func (b *bareMetalInventory) updateDhcpOrSNONetworkParams(updates map[string]int
 		setMachineNetworkCIDRForUpdate(updates, *machineCidr)
 		updates["api_vip"] = ""
 		updates["ingress_vip"] = ""
-		return network.VerifyMachineCIDR(swag.StringValue(params.ClusterUpdateParams.MachineNetworkCidr), cluster.Hosts, log)
+		mustHaveHosts := cluster.KubeKeyName == "" && cluster.KubeKeyNamespace == ""
+		return network.VerifyMachineCIDR(
+			swag.StringValue(params.ClusterUpdateParams.MachineNetworkCidr),
+			cluster.Hosts,
+			log,
+			mustHaveHosts)
```

Comment:
The parameter here is called mustHaveHosts, and in the function is kubeAPIMode.  It is confusing